src/main/java/bisq/core/dao/node/parser/TxParser.java

```diff
@@ -154,6 +251,36 @@ public TxParser(TxInputParser txInputParser,
             return Optional.empty();
     }
 
+    private boolean isFeeAndPhaseValid(int blockHeight, TempTx tempTx, long bsqFee, DaoPhase.Phase phase, Param param) {
+        // The leftover BSQ balance from the inputs is the BSQ fee in case we are in an OP_RETURN output
+
+        if (!isPhaseValid(blockHeight, tempTx, phase))
+            return false;
+
+        long paramValue = bsqStateService.getParamValue(param, blockHeight);
+        boolean isFeeCorrect = bsqFee == paramValue;
+        if (!isFeeCorrect) {
+            log.warn("Invalid fee. used fee={}, required fee={}", bsqFee, paramValue);
+            tempTx.setTxType(TxType.INVALID);
+            // TODO should we return already?
+
+            return false;
+        }
+
+        return true;
```

Comment:
Nit: Could do `return isFeeCorrect` on this line and remove the `return false` on L267.